CorsixTH/Lua/world.lua

```diff
@@ -1647,7 +1653,7 @@ function World:getIdleTile(x, y, idx)
     self.idle_cache[cache_idx] = cache
   end
   if not cache.x[idx] then
-    local ix, iy = self.pathfinder:findIdleTile(x, y, idx)
+    local ix, iy = self.pathfinder:findIdleTile(x, y, idx, parcel_id or 0)
```

Comment:
Is this a standard practice to put the `x or y` in the function parameter? Just asking

Reply:
If a function `f` wants value `x+1` rather than `x`, you write `f(x + 1)`.

If a function `g` wants value `x` or `0` if `x` is `nil` or `false` I would write `g(x or 0)`

In other words, `+` and `or` are the same thing to me, they compute a value desired by the function.
Why do you think `or` is different?